### PR TITLE
[ Fixes: #2 ] Training and Testing a Classification Model- Ensemble Method (Forests of Randomized Trees) on defaults.csv

### LINKED ISSUE #2: [Outreachy applications] Startup task: Train and test a classification model
This is a good way to get started with the environment and the problem domain. It will also provide the basis for a test case for future work. At a minimum, you should:

- load a dataset from the repo
- train a classification model from [scikit-learn](https://scikit-learn.org/stable/supervised_learning.html)
- compute an [evaluation metric](https://scikit-learn.org/stable/modules/model_evaluation.html) on a held-out test set

Feel free to include any additional steps you feel are relevant or you are interested in trying out, such as:

- basic exploratory analysis of the dataset
- data preprocessing
- hyperparameter tuning

When you start work on this task, please post a comment here indicating which dataset and model you will be working with so that other contributors can avoid duplicating your work.

### PR DESCRIPTION
**Pull request by archisha-chandel**
[ Fixes #2 ]
Ensemble method- **Forests of Randomized Trees** under 'sklearn' has been implemented on **defaults.csv**
It includes the following:
- Exploratory data analysis

- Data pre-processing

- Hyper-parameter tuning of the model (done manually through experimentation)

- Fitting and Prediction on train-test data

- Computing evaluating metrics

**Note**: In pre-processing KMeans clustering has also been implemented to achieve better results.

The classification report looks as follows:
image

`pandas_profiling` is an extra library that has been added to the environment and it is of great advantage which are stated below:

_For each column the following statistics - if relevant for the column type - are presented in an interactive HTML report:_

1. **Type inference**: detect the types of columns in a dataframe.
2. **Essentials**: type, unique values, missing values
3. **Quantile statistics** like minimum value, Q1, median, Q3, maximum, range, interquartile range
4. **Descriptive statistics** like mean, mode, standard deviation, sum, median absolute deviation, coefficient of variation, kurtosis, skewness
5. **Most frequent values**
6. **Histogram**
7. **Correlations** highlighting of highly correlated variables, Spearman, Pearson and Kendall matrices
8. **Missing values** matrix, count, heatmap and dendrogram of missing values
9. **Text analysis learn** about categories (Uppercase, Space), scripts (Latin, Cyrillic) and blocks (ASCII) of text data.

### CODE REVIEW

File: dev/archisha-chandel/defaults_modules.py [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+import pandas_profiling
+import matplotlib.pyplot as plt
+from sklearn.cluster import KMeans
+from sklearn.preprocessing import LabelEncoder
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import roc_auc_score,accuracy_score,\
+classification_report,roc_curve,confusion_matrix
+
+class DefineModules():
```
[the remark this answers is not in the document]
Comment (archisha-chandel) [reply dev/archisha-chandel/defaults_modules.py]: Requested changes have been made

**Comment by archisha-chandel** [comment]
I have added details and reasoning of the models, functions and hyper-parameters used.
